app.rb

```diff
@@ -0,0 +1,100 @@
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_accessor :people, :book, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_books
+    @books.each do |book|
+      puts "Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    @people.each do |person|
+      if person.is_a?(Student)
+        puts "[Student] Name: #{person.name}, ID: #{person.id} Age: #{person.age}"
+      elsif person.is_a?(Teacher)
+        puts "[Teacher] Name: #{person.name}, ID: #{person.id} Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    print 'Do you want to add a student (1) or a teacher (2)? [select number 1 or 2]:'
+    person_type = gets.chomp.to_i
+
+    print 'Age:'
+    age = gets.chomp
+
+    print 'Name:'
+    name = gets.chomp
+
+    if person_type == 2
+      print 'Specialization:'
+      gets.chomp
+      @people << Teacher.new(name, age, Specialization)
+    else
+      print 'Have Parent permission? [Y/N]'
+      parent_permission = gets.chomp.downcase == 'y'
+      @people << Student.new(name, age, parent_permission, Specialization)
+    end
```

Comment:
- [ ] There are error here `**uninitialized constant App::Specialization (NameError)**`. As I can see you haven't initialized specialization here. Please fix that.